EIPS/eip-2016.md

```diff
@@ -0,0 +1,241 @@
+  - A malicious or faulty endpoint could return arbitrary chain IDs, and potentially cause the user to sign transactions for unintended chains.
+- Verify that the specified chain ID matches the return value of `eth_chainId` from the endpoint.
+  - This should be done before any other interactions with the endpoint.
+
+### RPC Endpoints
+
+Wallets generally interact with chains via an RPC endpoint, identified by some URL.
+Most wallets ship with a set of chains and corresponding trusted RPC endpoints.
+The endpoints identified by the `rpcUrl` parameter cannot be assumed to be honest or even correct.
+Moreover, even trusted endpoints endpoints can expose users to privacy risks depending on their data collection practices.
+
+Therefore, the wallet should:
+
+- Inform users that their on-chain activity and IP address will be exposed to the endpoint.
+- If the endpoint is unknown to the wallet, inform users that the endpoint may behave in unexpected ways.
+- Observe good web security practices when interacting with the endpoint, such as require HTTPS.
+
+### UX
+
+Adding a new chain to the wallet can have significant implications for the functionality of the wallet and the experience of the user.
+A chain should never be added without the explicit consent of the user, and different chains should be clearly differentiated in the wallet UI.
+In service of these goals, the wallet should:
+
+- When receiving a `wallet_addEthereumChain` request, display a user confirmation informing the user that a specific dapp has requested that the chain be added.
+  - The confirmation used in [EIP-1102](./eip-1102.md) may serve as a point of reference.
+- Ensure that any chain metadata, such as `nativeCurrency` and `blockExplorerUrl`, are validated and used to maximum effect in the UI.
+- If the wallet UI has a concept of a "currently selected" or "currently active" chain, ensure that the user understands when a chain added using `wallet_addEthereumChain` becomes selected.
+
+### Validating Chain Data
+
+A wallet that implements `wallet_addEthereumChain` should expect to encounter requests for chains completely unknown to the wallet maintainers.
+That said, community resources exist that can be leveraged to verify requests for many Ethereum chains.
+The wallet should maintain a list of known chains, and verify requests add chains against that list, perhaps preferring data from the list over data submitted with a `wallet_addEthereumChain` request.
+
+At present, such community resources include:
+
+- [chainId.network](https://chainid.network)
+
+## References
+
+- [chainId.network](https://chainid.network)
+- Related EIPs
+  - [EIP-155: Simple replay attack protection](./eip-155.md)
+  - [EIP-695: Create `eth_chainId` method for JSON-RPC](./eip-695.md)
+  - [EIP-1102: Opt-in Account Exposure](./eip-1102.md)
+  - [EIP-2015: Wallet Update Chain JSON-RPC Method (`wallet_updateChain`)](./eip-2015.md)
+
```

Comment:
```suggestion
```
This is not an official EIP section, and we tend to avoid non-standard sections whenever possible.  I believe all of these are already linked inline, so this section doesn't provide any additional value to the EIP and the content probably should be removed.